docs/usage/getting-started/private-packages.md

```diff
@@ -398,6 +400,16 @@ npmRegistries:
     npmAuthToken: <Decrypted PAT Token>
 ```
 
+Thus, to allow for private registries to be used locally via the `NPM_TOKEN` environment variable _and_ with Renovate, your local `.yarnrc.yml` must contain:
```

Comment:
I am not sure this works, and this does not seem to be the recommendation from the linked discussion

Reply:
The discussion is confused, so I can understand why it wouldn't be clear.

https://github.com/renovatebot/renovate/issues/14756#issuecomment-1578651075

my experience is similar to the one link here, in an internal repo we have


.yarnrc.yml

```
nodeLinker: node-modules

npmRegistries:
  "//registry.yarnpkg.com/":
    npmAuthToken: "${NPM_TOKEN}"

```

and renovate.json

```

  "hostRules": [
    {
      "matchHost": "https://registry.yarnpkg.com/",
      "hostType": "npm",
      "encrypted": {
        "token": "[REDACTED]"
      }
    }
  ]
```

yarn locally works because the .yarnrc.yml, and it is able to pull the private registries fine in renovate.

The ${NPM_TOKEN:-} issue discussed is only relevant if you have a use of the NPM_TOKEN environment variable that is not replaced by the host matching rules, so I did not include it in the text.

This all works as long as the fuzzy matched merging performed at https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/npm/post-update/index.ts#L553-L559 correctly overwrites the required env var.

Does that make sense? Would you like me to try to publish a repro? (I'm not sure exactly what I would use a private registry)